test/system/src/main/java/io/pravega/test/system/framework/kubernetes/K8sClient.java

```diff
@@ -670,30 +671,63 @@ public CompletableFuture<V1Status> deleteSecret(String name, String namespace) {
      */
     public CompletableFuture<Void> waitUntilPodIsRunning(String namespace, String labelName, String labelValue, int expectedPodCount) {
 
-        AtomicBoolean shouldRetry = new AtomicBoolean(true);
+        return waitUntilPodIsRunningRetries(namespace, labelName, labelValue, expectedPodCount, POD_RUNNING_RETRIES);
+    }
+
+    /**
+     * A method which returns a CompletableFuture once *all* of the existing pods for the given label are in the running
+     * state. This means that it will wait for terminated pods to be removed in the case of a scale down event before
+     * proceeding.
+     *
+     * @param namespace Namespace
+     * @param labelName Label name.
+     * @param labelValue Value of the Label.
+     * @param expectedPodCount Number of pods that need to be running.
+     * @param retries The number of retries to attempt.
+     * @return A future which completes once the number of running pods matches the given criteria.
+     */
+    public CompletableFuture<Void> waitUntilPodIsRunningRetries(String namespace, String labelName, String labelValue, int expectedPodCount, int retries) {
+        AtomicInteger retriesRemaining = new AtomicInteger(retries);
+        Supplier<Boolean> shouldRetry = () -> retriesRemaining.getAndDecrement() > 0;
 
-        return Futures.loop(shouldRetry::get,
+        return Futures.loop(shouldRetry,
                 () -> Futures.delayedFuture(Duration.ofSeconds(5), executor) // wait for 5 seconds before checking for status.
-                        .thenCompose(v -> getStatusOfPodWithLabel(namespace, labelName, labelValue)) // fetch status of pods with the given label.
-                        .thenApply(podStatuses -> podStatuses.stream()
-                                // check for pods where all containers are running.
-                                .filter(podStatus -> {
-                                    if (podStatus.getContainerStatuses() == null) {
-                                        return false;
-                                    } else {
-                                        return podStatus.getContainerStatuses()
-                                                .stream()
-                                                .allMatch(st -> st.getState().getRunning() != null);
-                                    }
-                                }).count()),
-                runCount -> { // Number of pods which are running
-                    log.info("Expected running pod count of {}:{}, actual running pod count of {}:{}.", labelValue, expectedPodCount, labelValue, runCount);
-                    if (runCount == expectedPodCount) {
-                        shouldRetry.set(false);
+                        .thenCompose(v -> getPodsWithLabel(namespace, labelName, labelValue)) // fetch status of pods with the given label.
```

Comment:
is this change related to tls?

Reply:
This change is part of https://github.com/pravega/pravega/pull/6445